sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BaseBlobClientBuilder.java

```diff
@@ -23,6 +39,7 @@ protected final String getServiceUrlMidfix() {
 
     @Override
     protected final void applyServiceSpecificValidations(ResponseValidationPolicyBuilder builder) {
-        // for blob service validations
+        // CPK
+        builder.addOptionalEcho(Constants.HeaderConstants.ENCRYPTION_KEY_SHA256);
```

Comment:
This will require the recording framework to capture another header, any tests using this check will fail as that hasn't been updated.

Reply:
Fixed in PR #5297, will let you know when merged

Reply:
Merged